### PR TITLE
Websocket subprotocol

### PR DESCRIPTION
**Pull request by darkvariantdivine**
Fix for issue #1878, @ahopkins for your review

### CODE REVIEW

File: sanic/websocket.py [hunk 1]
```diff
@@ -137,9 +137,11 @@ def __init__(
         self,
         send: Callable[[ASIMessage], Awaitable[None]],
         receive: Callable[[], Awaitable[ASIMessage]],
+        subprotocols: List[str]=[],
```
Comment (ahopkins) [inline sanic/websocket.py:140]: This should default to `None`. If there needs to be a default list, that should be handled inside the function. [See rationale](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments)

```
subprotocols: Optional[List[str]]=None

...

subprotocols = subprotocols or []
```

File: tests/test_asgi.py [hunk 2]
```diff
@@ -208,6 +208,21 @@ async def test_websocket_receive(send, receive, message_stack):
...
+    subprotocols = ['graphql-ws', 'test']
+
+    ws = WebSocketConnection(send, receive, subprotocols)
+    await ws.accept()
+
+    assert len(message_stack) == 1
+
+    message = message_stack.popleft()
+    assert message["type"] == "websocket.accept"
+    assert message['subprotocol'] == "graphql-ws,test"
```
Comment (ahopkins) [inline tests/test_asgi.py:222]: Let's also add a test where there is only 1 subprotocol.

File: sanic/websocket.py (hunk 1 above)
Comment (darkvariantdivine) [reply sanic/websocket.py:140]: Updated, thanks for the advice!

File: tests/test_asgi.py (hunk 2 above)
Comment (darkvariantdivine) [reply tests/test_asgi.py:222]: Added test for single subprotocol and no subprotocols

**Comment by darkvariantdivine** [comment]
@ahopkins I fixed all the linter errors and made all the changes you requested :)

**Pull request merged**